orchagent/crmorch.cpp

```diff
@@ -240,7 +240,7 @@ void CrmOrch::handleSetCommand(const string& key, const vector<FieldValueTuple>&
             if (field == CRM_POLLING_INTERVAL)
             {
                 m_pollingInterval = chrono::seconds(to_uint<uint32_t>(value));
-                auto interv = timespec { .tv_sec = m_pollingInterval.count(), .tv_nsec = 0 };
+                auto interv = timespec { .tv_sec = (time_t)m_pollingInterval.count(), .tv_nsec = 0 };
```

Comment:
>(time_t)m_pollingInterval.count() [](start = 51, length = 33)

Use mktime(). time_t is not portable if directly used.

Reply:
time_t mktime(struct tm *tm) accepts "tm" and not the seconds.
As per code, m_pollingInterval.count() seems to be secs but was declared as ulong. time_t is also typedef as long in .h file.

Reply:
time_t is 32-bit signed in 32 bit gcc, and 64-bit sign in 64 bit gcc. So time_t is not portable if directly used in casting. You can first construct a `tm` with a integer sec field, and then use `mktime` to get a `time_t`.

---
In reply to: [309991958](https://github.com/Azure/sonic-swss/pull/1015#discussion_r309991958) [](ancestors = 309991958)

Reply:
Done

Reply:
Reverted the mktime changes, and kept typecasting implementation